corehq/apps/data_dictionary/util.py

```diff
@@ -62,16 +55,32 @@ def _get_current_case_types_and_properties(domain):
 
 
 def add_properties_to_data_dictionary(domain, case_type, properties):
-    case_type_obj, created = CaseType.objects.get_or_create(domain=domain, name=case_type)
-    if not created:
-        old_properties = set(CaseProperty.objects.filter(case_type=case_type_obj).values_list('name', flat=True))
-    else:
-        old_properties = set()
-    new_properties = []
-
-    for case_property in properties:
-        if case_property not in old_properties:
-            old_properties.add(case_property)
-            new_properties.append(CaseProperty(name=case_property, case_type=case_type_obj))
-
-    CaseProperty.objects.bulk_create(new_properties)
+    if not properties:
+        return
+
+    _create_properties_for_case_types(domain, {case_type: properties})
+
+
+def _create_properties_for_case_types(domain, case_type_to_prop):
+    current_case_types, current_properties = _get_current_case_types_and_properties(domain)
+    new_case_properties = []
+
+    for case_type, props in case_type_to_prop.items():
+        if not case_type:
+            continue
+
+        if case_type in current_case_types:
+            case_type_obj = current_case_types[case_type]
+        else:
+            case_type_obj = CaseType.objects.create(domain=domain, name=case_type)
+
+        for prop in props:
+            if '/' in prop:
```

Comment:
is this here because parent / host properties are referenced with `/` (e.g. `index/parent/case_prop`)? If so, might be good to either write a comment, or do something like `property_references_indexed_case = '/' in prop`